src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1103,31 +1103,48 @@
     (dissoc row-map :environment_id)
     row-map))
 
+(defn normalize-resource-event
+  "Prep `event` for comparison/computation of a hash"
+  [event]
+  (-> event
+      (utils/update-when [:timestamp] to-timestamp)
+      (utils/update-when [:old_value] sutils/db-serialize)
+      (utils/update-when [:new_value] sutils/db-serialize)
+      (assoc :containing_class (find-containing-class (get event :containment_path)))))
+
+(defn normalize-report
+  "Prep the report for comparison/computation of a hash"
+  [report]
+  (-> report
+      (update-in [:start_time] to-timestamp)
+      (update-in [:end_time] to-timestamp)
+      (update-in [:resource_events] #(map normalize-resource-event %))))
+
+
+(defn convert-containment-path
+  "Convert the contain path from a collection to the jdbc array type"
+  [event]
+  (utils/update-when event
+                     [:containment_path]
+                     (fn [cp]
+                       (when cp
```

Comment:
This when seems unnecessary unless you expect `cp` to be `false`. `update-when` should already only update if its non-nil.